pkg/envsupport/support.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package envsupport
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/support/kind"
+)
+
+type ContextKey string
+
+func (c ContextKey) String() string {
+	return string(c)
+}
+
+var (
+	// ContextKeyKubeConfig key for setting kube config value in context
+	ContextKeyKubeConfig = ContextKey("kubeconfig")
+	// ContextKeyClusterName key for setting clustername value in context
+	ContextKeyClusterName = ContextKey("clustername")
+)
+
+// GetValueFromContext gets the caller value from the context.
+// Use this function to retrieve value of a context key
+// or can print out value of key by doing,
+// fmt.Println("Key is:", ContextKeyKubeConfig.String())
+func GetValueFromContext(ctx context.Context, key ContextKey) (string, bool) {
```

Comment:
If we're going to provide a helper func to retrieve data in context, we cant assume that the value will be a string. This should be something like 
```go
func GetValueFromContext(ctx context.Context, key ContextKey) (interface{}, bool)
```
Or, rename it to indicate the type returned
```go
func GetStringValueFromContext(ctx, key ContextKey)(string, bool)
```